### PR TITLE
Copy jaeger functionality from java-spring-cloud

### PR DESCRIPTION
**Pull request by geoand**
I moved the core Jaeger auto-configuration functionality from `java-spring-cloud` into this repo.

This contains repo the bare minimum meaning that the module will not pull in any other OpenTracing auto-configurations dependencies.

@pavolloffay  @metacosm @cmoulliard Can you place a good look at this and let me know what needs to be changed?

@metacosm, @cmoulliard I would particularly like to know what you think the developer expected  should be for those that use this module.
@pavolloffay I don't have any experience with the release scripts or much experience with Travis, so I would appreciate your input

### CODE REVIEW

**Comment by pavolloffay** [comment]
Bintray repo https://bintray.com/opentracing/maven/opentracing-spring-jaeger/edit?tab=general

**Comment by geoand** [comment]
@pavolloffay Do you have to do anything as an admin for Travis? I'm not able to encrypt anything because I don't have permissions.

File: opentracing-spring-jaeger-autoconfigure/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.opentracing.contrib</groupId>
+        <artifactId>opentracing-spring-jaeger-parent</artifactId>
+        <version>0.1.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>opentracing-spring-jaeger-autoconfigure</artifactId>
```
Comment (pavolloffay) [inline opentracing-spring-jaeger-autoconfigure/pom.xml]: Can we name it `-starter`?

Comment (geoand) [reply opentracing-spring-jaeger-autoconfigure/pom.xml]: Sure

Comment (geoand) [reply opentracing-spring-jaeger-autoconfigure/pom.xml]: Done

File: README.md [hunk 2]
```diff
@@ -0,0 +1,161 @@
+[![Build Status][ci-img]][ci] [![Released Version][maven-img]][maven]
+
+## Configuration
+
+```xml
+<dependency>
+  <groupId>io.opentracing.contrib</groupId>
+  <artifactId>java-spring-jaeger-parent</artifactId>
```
Comment (pavolloffay) [inline README.md]: This seems wrong :/ :D

File: .mvn/wrapper/maven-wrapper.properties [hunk 3]
```diff
@@ -0,0 +1 @@
+distributionUrl=https://repo1.maven.org/maven2/org/apache/maven/apache-maven/3.3.9/apache-maven-3.3.9-bin.zip
```
Comment (pavolloffay) [inline .mvn/wrapper/maven-wrapper.properties]: You are using older wrapper version

Comment (pavolloffay) [reply .mvn/wrapper/maven-wrapper.properties]: New one can be pinned by `mvn -N io.takari:maven:wrapper -Dmaven=3.5.3`

File: README.md (hunk 2 above)
Comment (geoand) [reply README.md]: Fixed

File: .mvn/wrapper/maven-wrapper.properties (hunk 3 above)
Comment (geoand) [reply .mvn/wrapper/maven-wrapper.properties]: Fixed

**Comment by geoand** [comment]
@pavolloffay Have you been able to review the minor fixes?

**Comment by pavolloffay** [comment]
No but I think it definitely looks great!

**Comment by geoand** [comment]
:)

Ok then, I will squash, merge and attempt a release

**Pull request merged**